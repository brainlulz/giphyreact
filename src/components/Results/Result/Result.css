.ResultImage {
border: 0.5rem solid #FFBD39;
  width: 100%;
  height: 100%;
  z-index: 0;
}

.favorited {
  border: 0.5rem solid #E61C5D;
}

.Result:hover .Modal{
  opacity: 1;
}