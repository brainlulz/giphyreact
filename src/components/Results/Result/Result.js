import React from "react";
import classNames from 'classnames';
import "./Result.css";

import Modal from "../Modal/Modal";

const Result = ({ favorites, gifInfo, onToggleFavorite, favorited }) => (
  <div className="Result">
    <Modal
      onToggleFavorite={onToggleFavorite}
      hrefLink={gifInfo.url}
      gif={gifInfo}
    />
    <img
      className={classNames('ResultImage', {
        favorited,
      })}
      id={gifInfo.id}
      src={gifInfo.images.fixed_height_downsampled.url}
      alt={gifInfo.slug}
    />
  </div>
);

export default Result;
