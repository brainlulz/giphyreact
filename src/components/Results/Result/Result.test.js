import React from 'react';
import { shallow } from 'enzyme';
import Result from './Result';

describe('Result', () => {
  const gifInfo = {
    id: 3,
    url: 'url',
    images: { fixed_height_downsampled: { url: 'downsampled.gif' } }
  };
  it('renders the Result component', () => {
    const wrapper = shallow(<Result gifInfo={gifInfo} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders the Result component favorited', () => {
    const wrapper = shallow(<Result gifInfo={gifInfo} favorited/>);
    expect(wrapper).toMatchSnapshot();
  });
});
