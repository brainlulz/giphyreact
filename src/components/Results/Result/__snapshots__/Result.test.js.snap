// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result renders the Result component 1`] = `
<div
  className="Result"
>
  <Modal
    gif={
      Object {
        "id": 3,
        "images": Object {
          "fixed_height_downsampled": Object {
            "url": "downsampled.gif",
          },
        },
        "url": "url",
      }
    }
    hrefLink="url"
  />
  <img
    className="ResultImage"
    id={3}
    src="downsampled.gif"
  />
</div>
`;

exports[`Result renders the Result component favorited 1`] = `
<div
  className="Result"
>
  <Modal
    gif={
      Object {
        "id": 3,
        "images": Object {
          "fixed_height_downsampled": Object {
            "url": "downsampled.gif",
          },
        },
        "url": "url",
      }
    }
    hrefLink="url"
  />
  <img
    className="ResultImage favorited"
    id={3}
    src="downsampled.gif"
  />
</div>
`;
